neuralforecast/core.py

```diff
@@ -422,6 +451,99 @@ def cross_validation(
         fcsts_df = fcsts_df.merge(df, how="left", on=["unique_id", "ds"])
         return fcsts_df
 
+    def predict_insample(self, step_size: int = 1):
+        """Predict insample with core.NeuralForecast.
+
+        `core.NeuralForecast`'s `predict_insample` uses stored fitted `models`
+        to predict historic values of a time series from the stored dataframe.
+
+        Parameters
+        ----------
+        step_size : int (default=1)
+            Step size between each window.
+
+        Returns
+        -------
+        fcsts_df : pandas.DataFrame
+            DataFrame with insample predictions for all fitted `models`.
+        """
+        if not self._fitted:
+            raise Exception(
+                "The models must be fitted first with `fit` or `cross_validation`."
+            )
+
+        for model in self.models:
+            if model.SAMPLING_TYPE == "recurrent":
+                warnings.warn(
+                    f"Predict insample might not provide accurate predictions for \
+                       recurrent model {repr(model)} yet due to scaling."
+                )
+                print(
+                    f"WARNING: Predict insample might not provide accurate predictions for \
+                      recurrent model {repr(model)} yet due to scaling."
+                )
+
+        cols = []
```

Comment:
Add comment explaining that we are covering repeated model apparitions with a suffix.